lib/kubernetes-deploy/kubernetes_resource/cloudsql.rb

```diff
@@ -0,0 +1,66 @@
+module KubernetesDeploy
+  class Cloudsql < KubernetesResource
+    TIMEOUT = 30.seconds
+
+    def initialize(name, namespace, file)
+      @name, @namespace, @file = name, namespace, file
+    end
+
+    def sync
+      _, st = run_kubectl("get", type, @name)
+      @found = st.success?
+      @status = true
+
+
+
+      log_status
+    end
+
+    def deploy_succeeded?
+      cloudsql_proxy_deployment_exists? && mysql_service_exists?
+    end
+
+    def deploy_failed?
+      !cloudsql_proxy_deployment_exists? || !mysql_service_exists?
```

Comment:
If I'm not mistaken, these conditions mean that the deploy is incomplete, not that it failed. `deploy_failed?` should only return `true` when it the resource is doomed--it'll abort the Shipit deploy. As a result, several resource types have `false` hardcoded here. Does the cloudsql controller report anything if instance provisioning hard-fails for some reason (e.g. a instance count cap)?